restic/cli/restore.go

```diff
@@ -212,22 +212,28 @@ func (r *Restic) parsePath(paths []string) string {
 }
 
 func (r *Restic) s3Restore(log logr.Logger, snapshot Snapshot, stats *RestoreStats) error {
-
 	log.Info("S3 chosen as restore destination")
 
+	restoreS3Endpoint, s3AccessKey, s3SecretKey, err := r.getS3FromEnv()
+	if err != nil {
+		log.Error(err, "unable to load S3 information")
+		return err
+	}
+
 	snapDate := snapshot.Time.Format(time.RFC3339)
 	PVCName := r.parsePath(snapshot.Paths)
 	fileName := fmt.Sprintf("backup-%v-%v-%v.tar.gz", snapshot.Hostname, PVCName, snapDate)
-	stats.RestoreLocation = fmt.Sprintf("%s/%s", os.Getenv(RestoreS3EndpointEnv), fileName)
+
+	stats.RestoreLocation = fmt.Sprintf("%s/%s", restoreS3Endpoint, fileName)
 	stats.SnapshotID = snapshot.ID
 
-	s3Client := s3.New(os.Getenv(RestoreS3EndpointEnv), os.Getenv(RestoreS3AccessKeyIDEnv), os.Getenv(RestoreS3SecretAccessKey))
-	err := s3Client.Connect()
+	errorChannel, uploadWritePipe, err := r.startS3Upload(fileName, restoreS3Endpoint, s3AccessKey, s3SecretKey)
 	if err != nil {
 		return err
 	}
 
-	uploadReadPipe, uploadWritePipe := io.Pipe()
+	zipWriter := gzip.NewWriter(uploadWritePipe)
```

Comment:
If we're already touching this, I would prefer to call the variable **g**zipWriter since I was super confused by the `tar.Writer` later on.

zip != gzip